src/proxy-ui-api/src/main/java/org/niis/xroad/restapi/service/ServiceDescriptionService.java

```diff
@@ -534,6 +581,71 @@ public ServiceDescriptionType refreshServiceDescription(Long id, boolean ignoreW
         throw new NotImplementedException("REST ServiceDescription refresh not implemented yet");
     }
 
+    /**
+     * Refresh REST service description
+     *
+     * @param serviceDescriptionType
+     * @return {@link ServiceDescriptionType}
+     * @throws WrongServiceDescriptionTypeException if service type is not REST
+     * @throws ServiceNotFoundException             if there is no service linked to service description
+     * @throws InvalidUrlException                  if url is invalid
+     */
+    private ServiceDescriptionType refreshRESTServiceDescription(ServiceDescriptionType serviceDescriptionType)
+            throws WrongServiceDescriptionTypeException, ServiceNotFoundException, InvalidUrlException {
+        verifyAuthority("REFRESH_REST");
+
+        if (!serviceDescriptionType.getType().equals(DescriptionType.REST)) {
+            throw new WrongServiceDescriptionTypeException("Expected description type REST");
+        }
+
+        validateUrl(serviceDescriptionType.getUrl());
+
+        serviceDescriptionType.setRefreshedDate(new Date());
+
+        return serviceDescriptionType;
+    }
+
+    /**
+     * Refresh OPENAPI3 ServiceDescription
+     *
+     * @param serviceDescriptionType
+     * @param ignoreWarnings
+     * @return {@link ServiceDescriptionType}
+     * @throws WrongServiceDescriptionTypeException if service type is not openapi3
+     * @throws ServiceNotFoundException             if there is no service linked to given service description
+     * @throws UnhandledWarningsException           if unhandled warnings are found and ignoreWarnings if false
+     * @throws OpenApiParser.ParsingException       if parsing openapi3 description fails
+     * @throws InvalidUrlException                  if url is invalid
+     */
+    private ServiceDescriptionType refreshOPENAPI3ServiceDescription(ServiceDescriptionType serviceDescriptionType,
```

Comment:
`refreshOPENAPI3ServiceDescription` differs from convention used elsewhere, should probably be `refreshOpenApi3ServiceDescription`? There is also variant `parseOpenapi3ToServiceDescription` in use in several places.

Reply:
Added fixes to align convention

Reply:
There seems to be one more still: `ServiceDescriptionService.addOpenapi3ServiceDescription`

Then some in tests, but those are not so important.